tests/e2e/kubetest2-kops/deployer/up.go

```diff
@@ -178,6 +182,21 @@ func (d *deployer) createCluster(zones []string, adminAccess string, yes bool) e
 		args = appendIfUnset(args, "--gce-service-account", "default")
 		args = appendIfUnset(args, "--networking", "kubenet")
 
+		// kubeapi server config https://github.com/kubernetes/kubernetes/blob/622509830c1038535e539f7d364f5cd7c3b38791/cluster/gce/gci/configure-kubeapiserver.sh#L106
+		for _, v := range strings.Split(d.KubeAPIServerAdmissionPlugins, ",") {
+			args = append(args, "--set", fmt.Sprintf("spec.kubeAPIServer.enableAdmissionPlugins=%s", v))
+		}
+		if strings.Contains(d.KubernetesFeatureGates, "AllAlpha") {
+			d.KubeAPIServerRuntimeConfig = "api/all=true"
+		}
+		for _, v := range strings.Split(d.KubeAPIServerRuntimeConfig, ",") {
+			args = append(args, "--set", fmt.Sprintf("spec.kubeAPIServer.runtimeConfig=%s", v))
+		}
+
+		args = append(args, "--set", "spec.kubeAPIServer.enableAggregatorRouting=true")
+		args = append(args, "--set", "spec.kubeAPIServer.auditLogPath=/var/log/kube-apiserver-audit.log")
+		args = append(args, "--set", "spec.kubeAPIServer.auditLogMaxSize=2000000000")
+		args = append(args, "--set", "spec.kubeAPIServer.logLevel=4")
```

Comment:
Why do all these have to be added here and not in test-infra, or some scenario like for scalability?
https://github.com/kubernetes/kops/blob/master/tests/e2e/scenarios/scalability/run-test.sh

Reply:
because kubeup sets many defaults at https://github.com/kubernetes/kubernetes/blob/67733c650ba69db2dc2a9b7e07df2e4f3adae7b5/cluster/gce/gci/configure-kubeapiserver.sh#L106 and I don't think I should be tweaking the defaults kops has been setting for a few releases

Reply:
I meant why not add them specifically to the tests, in test-infra repo, and not in kubetest deployer?
Probably some of these like the audit logs would be a good idea to add for all tests and also collect them.

Reply:
cool

Reply:
Done https://github.com/kubernetes/test-infra/pull/30931

This can be merged now.